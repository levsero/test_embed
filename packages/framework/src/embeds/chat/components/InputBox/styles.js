import styled, { css } from 'styled-components'
import { zdColorGrey300 } from '@zendeskgarden/css-variables'
import { Label, Textarea } from '@zendeskgarden/react-forms'
import { isMobileBrowser } from 'src/util/devices'

export const HiddenLabel = styled(Label)`
  position: absolute !important;
  overflow: hidden !important;
  width: 1px !important;
  height: 1px !important;
  padding: 0 !important;
  border: 0 !important;
  clip: rect(1px, 1px, 1px, 1px) !important;
`
export const Container = styled.div`
  vertical-align: bottom !important;
  margin: 0 ${(props) => 1 / props.theme.fontSize}rem;
`

export const StyledTextarea = styled(Textarea)`
  ${isMobileBrowser() &&
  css`
    padding: ${(props) => 8 / props.theme.fontSize}rem;
    border: none !important;
    border-radius: 0;
    box-shadow: none !important;
    font-size: ${(props) => 16 / props.theme.fontSize}rem !important;
  `}
  &::-webkit-scrollbar {
    width: ${(props) => 4 / props.theme.fontSize}rem;
  }

  &::-webkit-scrollbar-thumb {
    background: ${zdColorGrey300};
    border-radius: ${(props) => 4 / props.theme.fontSize}rem;
  }
`
