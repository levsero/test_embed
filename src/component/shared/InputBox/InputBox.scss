@import '../../../styles/_vars.scss';

.fieldDisabled {
  background: $zd-color-white-smoke;
  box-shadow: none !important;
}

.label {
  composes: u-isHiddenVisually from global;
}

.textField {
  margin-bottom: 0 !important;

  .fieldMobile {
    padding: #{12 / $font-size}rem 0;
    border: none !important;
    border-radius: 0;
    box-shadow: none !important;
  }

  .input {
    padding-left: #{18 / $font-size}rem;
    &::-webkit-scrollbar {
      width: #{4 / $font-size}rem;
    }

    &::-webkit-scrollbar-thumb {
      background: $zd-color-grey-300;
      border-radius: #{4 / $font-size}rem;
    }
  }

  .inputMobile {
    font-size: #{16 / $font-size}rem !important;
  }
}
