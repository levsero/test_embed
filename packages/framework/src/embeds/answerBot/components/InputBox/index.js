import PropTypes from 'prop-types'
import { Field } from '@zendeskgarden/react-forms'
import { TEST_IDS } from 'src/constants/shared'
import { keyCodes } from 'src/util/keyboard'
import { Container, HiddenLabel, Input } from './styles'

const InputBox = ({
  inputValue,
  name,
  placeholder,
  handleSendInputValue,
  updateInputValue,
  isMobile,
  onFocus,
  onBlur,
}) => {
  const handleKeyDown = (e) => {
    if (e.keyCode === keyCodes.ENTER && !e.shiftKey) {
      e.preventDefault()
      handleSendInputValue()
    }
  }

  const handleInputValueChanged = (e) => {
    const { value } = e.target
    updateInputValue(value)
  }

  return (
    <Container>
      <Field>
        <HiddenLabel>{placeholder}</HiddenLabel>
        <Input
          value={inputValue}
          onChange={handleInputValueChanged}
          onKeyDown={handleKeyDown}
          placeholder={placeholder}
          name={name}
          rows={isMobile ? 1 : 3}
          data-testid={TEST_IDS.MESSAGE_FIELD}
          onFocus={onFocus}
          onBlur={onBlur}
        />
      </Field>
    </Container>
  )
}

InputBox.propTypes = {
  inputValue: PropTypes.string,
  name: PropTypes.string.isRequired,
  placeholder: PropTypes.string.isRequired,
  handleSendInputValue: PropTypes.func.isRequired,
  updateInputValue: PropTypes.func,
  isMobile: PropTypes.bool,
  onFocus: PropTypes.func,
  onBlur: PropTypes.func,
}

export default InputBox
