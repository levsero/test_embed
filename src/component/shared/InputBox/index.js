import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { TextField, Label, Textarea } from '@zendeskgarden/react-textfields';

import { keyCodes } from 'utility/keyboard';

import { locals as styles } from './InputBox.scss';
import classNames from 'classnames';

export class InputBox extends Component {
  static propTypes = {
    inputValue: PropTypes.string,
    name: PropTypes.string.isRequired,
    placeholder: PropTypes.string.isRequired,
    handleSendInputValue: PropTypes.func.isRequired,
    updateInputValue: PropTypes.func,
    disabled: PropTypes.bool,
    isMobile: PropTypes.bool
  };

  static defaultProps = {
    inputValue: '',
    disabled: false,
    updateInputValue: () => {},
    isMobile: false
  };

  handleKeyDown = (e) => {
    if (e.keyCode === keyCodes.ENTER && !e.shiftKey) {
      e.preventDefault();
      this.props.handleSendInputValue();
    }
  }

  handleInputValueChanged = (e) => {
    const { value } = e.target;

    this.props.updateInputValue(value);
  }

  render = () => {
    const { placeholder, name, inputValue, disabled, isMobile } = this.props;
    const fieldClasses = classNames(styles.textField, {
      [styles.fieldDisabled]: disabled
    });
    const inputClasses = classNames(
      styles.input,
      {
        [styles.inputMobile]: this.props.isMobile,
        [styles.fieldMobile]: this.props.isMobile,
        [styles.inputDisabled]: disabled
      }
    );

    return (
      <TextField className={fieldClasses}>
        <Label className={styles.label}>
          {placeholder}
        </Label>
        <Textarea
          value={inputValue}
          disabled={disabled}
          onChange={this.handleInputValueChanged}
          onKeyDown={this.handleKeyDown}
          className={inputClasses}
          placeholder={placeholder}
          name={name}
          rows={isMobile ? 1 : 3} />
      </TextField>
    );
  };
}
