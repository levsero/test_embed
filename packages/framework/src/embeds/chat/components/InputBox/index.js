import PropTypes from 'prop-types'
import { Field } from '@zendeskgarden/react-forms'
import { TEST_IDS } from 'src/constants/shared'
import useTranslate from 'src/hooks/useTranslate'
import { keyCodes } from 'src/util/keyboard'
import { Container, HiddenLabel, StyledTextarea } from './styles'

const InputBox = ({
  currentMessage = '',
  sendChat,
  handleChatBoxChange,
  isMobile,
  onFocus,
  onBlur,
}) => {
  const translate = useTranslate()

  const handleKeyDown = (e) => {
    if (e.key === keyCodes.ENTER && !e.shiftKey) {
      e.preventDefault()
      sendChat()
    }
  }

  const handleChange = (e) => {
    const { value } = e.target
    handleChatBoxChange(value)
  }

  const placeholder = translate('embeddable_framework.chat.chatBox.placeholder.typeMessageHere_v2')

  return (
    <Container>
      <Field>
        <HiddenLabel>{placeholder}</HiddenLabel>
        <StyledTextarea
          value={currentMessage}
          onChange={handleChange}
          onKeyDown={handleKeyDown}
          placeholder={placeholder}
          name="chatBox"
          rows={isMobile ? 1 : 3}
          data-testid={TEST_IDS.MESSAGE_FIELD}
          onFocus={onFocus}
          onBlur={onBlur}
        />
      </Field>
    </Container>
  )
}

InputBox.propTypes = {
  isMobile: PropTypes.bool.isRequired,
  currentMessage: PropTypes.string,
  sendChat: PropTypes.func.isRequired,
  handleChatBoxChange: PropTypes.func.isRequired,
  onFocus: PropTypes.func,
  onBlur: PropTypes.func,
}

export default InputBox
