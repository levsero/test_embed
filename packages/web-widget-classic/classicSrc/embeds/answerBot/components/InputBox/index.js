import { TEST_IDS } from 'classicSrc/constants/shared'
import { keyCodes } from 'classicSrc/util/keyboard'
import PropTypes from 'prop-types'
import { useState } from 'react'
import { Field } from '@zendeskgarden/react-forms'
import { Container, HiddenLabel, Input } from './styles'

const InputBox = ({
  inputValue,
  name,
  placeholder,
  handleSendInputValue,
  questionValueChanged,
  isMobile,
  onFocus,
  onBlur,
}) => {
  // event.nativeEvent.isComposing and isComposing flag are used
  // to support composing text in foreign languages across browsers.
  const [isComposing, setIsComposing] = useState(false)
  const handleKeyUp = () => setIsComposing(false)
  const handleCompositionEnd = () => setIsComposing(true)

  const handleKeyDown = (e) => {
    if (e.key === keyCodes.ENTER && !e.shiftKey && !e.nativeEvent.isComposing && !isComposing) {
      e.preventDefault()
      handleSendInputValue()
    }
  }

  const handleInputValueChanged = (e) => {
    const { value } = e.target

    questionValueChanged(value)
  }

  return (
    <Container>
      <Field>
        <HiddenLabel>{placeholder}</HiddenLabel>
        <Input
          value={inputValue}
          onChange={handleInputValueChanged}
          onKeyDown={handleKeyDown}
          onKeyUp={handleKeyUp}
          onCompositionEnd={handleCompositionEnd}
          placeholder={placeholder}
          name={name}
          rows={isMobile ? 1 : 3}
          data-testid={TEST_IDS.MESSAGE_FIELD}
          onFocus={onFocus}
          onBlur={onBlur}
        />
      </Field>
    </Container>
  )
}

InputBox.propTypes = {
  inputValue: PropTypes.string,
  name: PropTypes.string.isRequired,
  placeholder: PropTypes.string.isRequired,
  handleSendInputValue: PropTypes.func.isRequired,
  questionValueChanged: PropTypes.func,
  isMobile: PropTypes.bool,
  onFocus: PropTypes.func,
  onBlur: PropTypes.func,
}

export default InputBox
