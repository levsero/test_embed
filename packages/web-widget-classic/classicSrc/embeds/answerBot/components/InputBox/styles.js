import { FONT_SIZE } from 'classicSrc/constants/shared'
import styled from 'styled-components'
import { Label, Textarea } from '@zendeskgarden/react-forms'
import { isMobileBrowser } from '@zendesk/widget-shared-services'

const Container = styled.div`
  vertical-align: bottom !important;
  margin: 0 ${1 / FONT_SIZE}rem;
`

const HiddenLabel = styled(Label)`
  position: absolute;
  overflow: hidden;
  width: 1px;
  height: 1px;
  padding: 0;
  border: 0;
  clip: rect(1px, 1px, 1px, 1px);
`

const inputStyle = (isMobile) => {
  if (isMobile) {
    return `
      padding: ${8 / FONT_SIZE}rem 0;
      border: none !important;
      border-radius: 0 !important;
      box-shadow: none !important;
      font-size: ${16 / FONT_SIZE}rem;

      &&&:focus {
        border: none !important;
        box-shadow: none !important;
      }
    `
  } else {
    return ''
  }
}

const Input = styled(Textarea)`
  ${inputStyle(isMobileBrowser())}
`

export { Container, HiddenLabel, Input }
