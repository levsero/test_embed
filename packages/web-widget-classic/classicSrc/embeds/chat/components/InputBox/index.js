import { TEST_IDS } from 'classicSrc/constants/shared'
import useTranslate from 'classicSrc/hooks/useTranslate'
import { keyCodes } from 'classicSrc/util/keyboard'
import PropTypes from 'prop-types'
import { useState } from 'react'
import { Field } from '@zendeskgarden/react-forms'
import { Container, HiddenLabel, StyledTextarea } from './styles'

const InputBox = ({
  currentMessage = '',
  sendChat,
  handleChatBoxChange,
  isMobile,
  onFocus,
  onBlur,
}) => {
  const translate = useTranslate()

  // event.nativeEvent.isComposing and isComposing flag are used
  // to support composing text in foreign languages across browsers.
  const [isComposing, setIsComposing] = useState(false)
  const handleKeyUp = () => setIsComposing(false)
  const handleCompositionEnd = () => setIsComposing(true)

  const handleKeyDown = (e) => {
    if (e.key === keyCodes.ENTER && !e.shiftKey && !e.nativeEvent.isComposing && !isComposing) {
      e.preventDefault()
      sendChat()
    }
  }

  const handleChange = (e) => {
    const { value } = e.target
    handleChatBoxChange(value)
  }

  const placeholder = translate('embeddable_framework.chat.chatBox.placeholder.typeMessageHere_v2')

  return (
    <Container>
      <Field>
        <HiddenLabel>{placeholder}</HiddenLabel>
        <StyledTextarea
          value={currentMessage}
          onChange={handleChange}
          onKeyDown={handleKeyDown}
          onKeyUp={handleKeyUp}
          onCompositionEnd={handleCompositionEnd}
          placeholder={placeholder}
          name="chatBox"
          rows={isMobile ? 1 : 3}
          data-testid={TEST_IDS.MESSAGE_FIELD}
          onFocus={onFocus}
          onBlur={onBlur}
        />
      </Field>
    </Container>
  )
}

InputBox.propTypes = {
  isMobile: PropTypes.bool.isRequired,
  currentMessage: PropTypes.string,
  sendChat: PropTypes.func.isRequired,
  handleChatBoxChange: PropTypes.func.isRequired,
  onFocus: PropTypes.func,
  onBlur: PropTypes.func,
}

export default InputBox
