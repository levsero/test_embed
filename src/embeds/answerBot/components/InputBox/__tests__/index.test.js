import { render, fireEvent } from '@testing-library/react'
import React from 'react'

import InputBox from '../index'

const renderComponent = (props = {}) => {
  const defaultProps = {
    name: 'inputBox',
    placeholder: 'placeholder',
    handleSendInputValue: noop
  }

  const componentProps = {
    ...defaultProps,
    ...props
  }

  return render(<InputBox {...componentProps} />)
}

test('renders the expected elements', () => {
  const { queryByPlaceholderText, queryByLabelText } = renderComponent()

  expect(queryByLabelText('placeholder')).toBeInTheDocument()
  expect(queryByPlaceholderText('placeholder')).toBeInTheDocument()
})

test('text area has 3 rows when in desktop', () => {
  const { getByPlaceholderText } = renderComponent()

  expect(getByPlaceholderText('placeholder').rows).toEqual(3)
})

test('text area has 1 row when in mobile', () => {
  const { getByPlaceholderText } = renderComponent({ isMobile: true })

  expect(getByPlaceholderText('placeholder').rows).toEqual(1)
})

test('renders with value', () => {
  const { getByPlaceholderText } = renderComponent({
    inputValue: 'doremi'
  })

  expect(getByPlaceholderText('placeholder').value).toEqual('doremi')
})

describe('events', () => {
  it('updates handler on change', () => {
    const handler = jest.fn()
    const { getByPlaceholderText } = renderComponent({
      updateInputValue: handler,
      placeholder: 'here'
    })

    fireEvent.change(getByPlaceholderText('here'), { target: { value: 'new' } })
    expect(handler).toHaveBeenCalledWith('new')
  })

  it('updates handler on enter key event', () => {
    const handler = jest.fn()
    const { getByPlaceholderText } = renderComponent({
      handleSendInputValue: handler,
      placeholder: 'here'
    })

    fireEvent.keyDown(getByPlaceholderText('here'), {
      key: 'Enter',
      keyCode: 13,
      which: 13
    })
    expect(handler).toHaveBeenCalled()
  })

  it('does not update handler on enter key event when shift key is pressed', () => {
    const handler = jest.fn()
    const { getByPlaceholderText } = renderComponent({
      handleSendInputValue: handler,
      placeholder: 'here'
    })

    fireEvent.keyDown(getByPlaceholderText('here'), {
      key: 'Enter',
      keyCode: 13,
      which: 13,
      shiftKey: true
    })
    expect(handler).not.toHaveBeenCalled()
  })
})
