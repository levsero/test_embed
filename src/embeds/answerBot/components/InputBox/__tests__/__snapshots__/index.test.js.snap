// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders as disabled 1`] = `
.c0 {
  vertical-align: bottom !important;
  margin: 0 0.07142857142857142rem;
}

.c1 {
  position: absolute;
  overflow: hidden;
  width: 1px;
  height: 1px;
  padding: 0;
  border: 0;
  -webkit-clip: rect(1px,1px,1px,1px);
  clip: rect(1px,1px,1px,1px);
}

<div>
  <div
    class="c0"
  >
    <div
      class="text__c-txt___Uzv-G range__c-range___1aYaS "
    >
      <label
        class="c1 text__c-txt__label___3wMPZ "
        data-garden-container-id="containers.field"
        data-garden-container-version="1.3.3"
        data-garden-id="forms.text_label"
        data-garden-version="7.1.10"
        for="2val-field_1.3.3--input"
        id="2val-field_1.3.3--label"
      >
        placeholder
      </label>
      <textarea
        aria-describedby="2val-field_1.3.3--hint"
        aria-invalid="false"
        aria-labelledby="2val-field_1.3.3--label"
        class="-textarea text__c-txt__input___38pRd text__c-txt__input--area___9Qxj1 "
        data-garden-id="forms.textarea"
        data-garden-version="7.1.10"
        data-testid="message-field"
        id="2val-field_1.3.3--input"
        name="inputBox"
        placeholder="placeholder"
        rows="3"
      />
    </div>
  </div>
</div>
`;

exports[`renders as mobile 1`] = `
.c0 {
  vertical-align: bottom !important;
  margin: 0 0.07142857142857142rem;
}

.c1 {
  position: absolute;
  overflow: hidden;
  width: 1px;
  height: 1px;
  padding: 0;
  border: 0;
  -webkit-clip: rect(1px,1px,1px,1px);
  clip: rect(1px,1px,1px,1px);
}

<div>
  <div
    class="c0"
  >
    <div
      class="text__c-txt___Uzv-G range__c-range___1aYaS "
    >
      <label
        class="c1 text__c-txt__label___3wMPZ "
        data-garden-container-id="containers.field"
        data-garden-container-version="1.3.3"
        data-garden-id="forms.text_label"
        data-garden-version="7.1.10"
        for="3val-field_1.3.3--input"
        id="3val-field_1.3.3--label"
      >
        placeholder
      </label>
      <textarea
        aria-describedby="3val-field_1.3.3--hint"
        aria-invalid="false"
        aria-labelledby="3val-field_1.3.3--label"
        class="-textarea text__c-txt__input___38pRd text__c-txt__input--area___9Qxj1 "
        data-garden-id="forms.textarea"
        data-garden-version="7.1.10"
        data-testid="message-field"
        id="3val-field_1.3.3--input"
        name="inputBox"
        placeholder="placeholder"
        rows="1"
      />
    </div>
  </div>
</div>
`;

exports[`renders the expected classes 1`] = `
.c0 {
  vertical-align: bottom !important;
  margin: 0 0.07142857142857142rem;
}

.c1 {
  position: absolute;
  overflow: hidden;
  width: 1px;
  height: 1px;
  padding: 0;
  border: 0;
  -webkit-clip: rect(1px,1px,1px,1px);
  clip: rect(1px,1px,1px,1px);
}

<div>
  <div
    class="c0"
  >
    <div
      class="text__c-txt___Uzv-G range__c-range___1aYaS "
    >
      <label
        class="c1 text__c-txt__label___3wMPZ "
        data-garden-container-id="containers.field"
        data-garden-container-version="1.3.3"
        data-garden-id="forms.text_label"
        data-garden-version="7.1.10"
        for="1val-field_1.3.3--input"
        id="1val-field_1.3.3--label"
      >
        placeholder
      </label>
      <textarea
        aria-describedby="1val-field_1.3.3--hint"
        aria-invalid="false"
        aria-labelledby="1val-field_1.3.3--label"
        class="-textarea text__c-txt__input___38pRd text__c-txt__input--area___9Qxj1 "
        data-garden-id="forms.textarea"
        data-garden-version="7.1.10"
        data-testid="message-field"
        id="1val-field_1.3.3--input"
        name="inputBox"
        placeholder="placeholder"
        rows="3"
      />
    </div>
  </div>
</div>
`;

exports[`renders with value 1`] = `
.c0 {
  vertical-align: bottom !important;
  margin: 0 0.07142857142857142rem;
}

.c1 {
  position: absolute;
  overflow: hidden;
  width: 1px;
  height: 1px;
  padding: 0;
  border: 0;
  -webkit-clip: rect(1px,1px,1px,1px);
  clip: rect(1px,1px,1px,1px);
}

<div>
  <div
    class="c0"
  >
    <div
      class="text__c-txt___Uzv-G range__c-range___1aYaS "
    >
      <label
        class="c1 text__c-txt__label___3wMPZ "
        data-garden-container-id="containers.field"
        data-garden-container-version="1.3.3"
        data-garden-id="forms.text_label"
        data-garden-version="7.1.10"
        for="4val-field_1.3.3--input"
        id="4val-field_1.3.3--label"
      >
        placeholder
      </label>
      <textarea
        aria-describedby="4val-field_1.3.3--hint"
        aria-invalid="false"
        aria-labelledby="4val-field_1.3.3--label"
        class="-textarea text__c-txt__input___38pRd text__c-txt__input--area___9Qxj1 "
        data-garden-id="forms.textarea"
        data-garden-version="7.1.10"
        data-testid="message-field"
        id="4val-field_1.3.3--input"
        name="inputBox"
        placeholder="placeholder"
        rows="3"
      >
        doremi
      </textarea>
    </div>
  </div>
</div>
`;
